packages/unmock-core/src/matcher.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  IMock,
+  IMockRequest,
+  ISerializedRequest,
+  ISerializedResponse,
+} from "./interfaces";
+
+type RequestComparator = (
+  intercepted: ISerializedRequest,
+  mock: IMockRequest,
+) => boolean;
+
+function propertyMatches({
+  property,
+  interceptedRequest,
+  mockRequest,
+}: {
+  property: keyof (ISerializedRequest & IMockRequest);
+  interceptedRequest: ISerializedRequest;
+  mockRequest: IMockRequest;
+}): boolean {
+  const mockRequestProperty = mockRequest[property];
+  const interceptedRequestProperty = interceptedRequest[property];
+
+  if (mockRequestProperty === undefined) {
+    return true;
+  }
+
+  if (mockRequestProperty instanceof RegExp) {
```

Comment:
Since we added RegExp, it could also be nice to add a glob-like pattern (with a simple `*` being replaced with `.*?` for a `RegExp` pattern)

Reply:
Yes that's a good idea, but I think it's simplest if the matcher for now works with regex and any globs are handled before the matcher by, e.g., [glob-to-regexp](https://www.npmjs.com/package/glob-to-regexp).

Reply:
Yes of course, it should be before the matcher. I'm not a fan of how `glob-to-regexp` does things though (I prefer lazy quantifiers over greedy ones in general - I think it captures the intent of `glob` better). We could write our own IMO in this case (future PR?)